ashpd-demo/src/portals/desktop/global_shortcuts.rs

```diff
@@ -0,0 +1,279 @@
+        pub activations: Arc<Mutex<HashSet<String>>>,
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for GlobalShortcutsPage {
+        const NAME: &'static str = "GlobalShortcutsPage";
+        type Type = super::GlobalShortcutsPage;
+        type ParentType = PortalPage;
+
+        fn class_init(klass: &mut Self::Class) {
+            klass.bind_template();
+
+            klass.install_action_async("global_shortcuts.start_session", None, |page, _, _| async move {
+                if let Err(err) = page.start_session().await {
+                    tracing::error!("Failed to request {}", err);
+                }
+            });
+            klass.install_action_async("global_shortcuts.stop", None, |page, _, _| async move {
+                page.stop().await;
+            });
+        }
+
+        fn instance_init(obj: &glib::subclass::InitializingObject<Self>) {
+            obj.init_template();
+        }
+    }
+    impl ObjectImpl for GlobalShortcutsPage {
+        fn constructed(&self) {
+            self.parent_constructed();
+            self.obj().action_set_enabled("global_shortcuts.stop", false);
+        }
+    }
+    impl WidgetImpl for GlobalShortcutsPage {}
+    impl BinImpl for GlobalShortcutsPage {}
+    impl PortalPageImpl for GlobalShortcutsPage {}
+}
+
+glib::wrapper! {
+    pub struct GlobalShortcutsPage(ObjectSubclass<imp::GlobalShortcutsPage>)
+        @extends gtk::Widget, adw::Bin;
+}
+
+impl GlobalShortcutsPage {
+    async fn start_session(&self) -> ashpd::Result<()> {
+        let root = self.native().unwrap();
+        let imp = self.imp();
+        let identifier = WindowIdentifier::from_native(&root).await;
+        let shortcuts = imp.shortcuts.text();
+        let shortcuts: Option<Vec<_>> = shortcuts.as_str().split(',')
+            .map(|desc| {
+                let mut split = desc.splitn(3, ':');
+                let name = split.next()?;
+                let desc = split.next()?;
+                let trigger = split.next();
+                Some(NewShortcut::new(name, desc).preferred_trigger(trigger))
+            }).collect();
+
+        match shortcuts {
+            Some(shortcuts) => {
+                let global_shortcuts = GlobalShortcuts::new().await?;
+                let session = global_shortcuts.create_session().await?;
+                let request = global_shortcuts.bind_shortcuts(&session, &shortcuts[..], &identifier).await?;
+                imp.response_group.set_visible(true);
+                let response = request.response();
+                imp.session_state_label.set_text(
+                    &match &response {
+                        Ok(_) => "OK".into(),
+                        Err(ashpd::Error::Response(ResponseError::Cancelled)) => "Cancelled".into(),
+                        Err(ashpd::Error::Response(ResponseError::Other)) => "Other response error".into(),
+                        Err(e) => format!("{}", e),
+                    }
+                );
```

Comment:
Same here, use the notifications infrastructure

Reply:
Is there a way to make the notifications stay? By the time I get back from the debugger, the message is already gone and I don't know what went wrong.

Reply:
that is a good point, but there isn't actually. I can change the notification widget implementation separately later :)